controllers/servicebinding_controller_test.go

```diff
@@ -516,6 +516,18 @@ var _ = Describe("ServiceBinding controller", func() {
 			})
 		})
 
+		Context("useMetaName annotation is provided", func() {
+			It("should put in the secret.instance_name the instance meta.name", func() {
+				createdInstance.Annotations = map[string]string{
+					api.UseInstanceMetaNameInSecret: "true",
+				}
+				updateInstance(ctx, createdInstance)
+				createdBinding = createBinding(ctx, bindingName, bindingTestNamespace, instanceName, "", "")
+				bindingSecret := getSecret(ctx, createdBinding.Spec.SecretName, createdBinding.Namespace, true)
+				validateSecretData(bindingSecret, "instance_name", instanceName)
```

Comment:
move validateInstanceInfo and validateSecretMetadata out of the "sync" context - create them as functions next to validateSecretData. 
Then in this test do  validateInstanceInfo and validateSecretMetadata. You want to make sure your changes (and future changes to that area) don't affect other unintended properties.

Reply:
done